### PR TITLE
fix: Zoho Bigin App update (API version V1 to V2, bug fix, additional information added to events)

### PR DESCRIPTION
**Pull request by sumandas-dev**
## What does this PR do?

1. Updated the Zoho Bigin API version (V1 -> V2)
2. Previously the app was creating contact for all the attendees which caused nameless contacts for the Guest attendees. Updated the app to create contact only for the main attendee
3. While creating contact and event, now it will include a tag `Calcom`. It will help the user to distinguish the contacts and events created from Cal.com
4. Previously, the app was only adding Additional notes and Locations to Bigin events. Now it will include Guests list, Meeting agenda, user created questions, additional notes, and Reschedule Reason.
5. Added the required environment variable to .env.appStore.example

#### Bug Fix

1. If a contact is already available in Zoho Bigin then if the same contact book an event on Cal.com, it does not add an event to Bigin.
2. Location field was not syncing although the app is sending it to Zoho Bigin.

These 2 bugs are fixed in this pull request

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Are there environment variables that should be set?
Two environment variable needs to be set in .env.appStore `ZOHO_BIGIN_CLIENT_ID ZOHO_BIGIN_CLIENT_SECRET`

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by PeerRich** [comment]
thank you!

File: packages/app-store/zoho-bigin/lib/CalendarService.ts [hunk 1]
```diff
@@ -163,6 +163,49 @@ export default class BiginCalendarService implements Calendar {
       .catch((e) => this.log.error("Error searching contact:", JSON.stringify(e), e.response?.data));
   }
 
+  /**
+   * Generates a description from the event data which includes the Agenda,Reschedule Reason, additional notes, attendees, and user fields.
+   */
+  private getDescription(event: CalendarEvent) {
```
Comment (joeauyeung) [inline packages/app-store/zoho-bigin/lib/CalendarService.ts:169]: **Blocking**: I think we need to revisit this function as well. I'm getting an empty string as the description

**Review by joeauyeung — CHANGES_REQUESTED** [review]
Thanks for your contribution @sumandas-dev. 

 I left some comments that should be addressed before approving.

File: packages/app-store/zoho-bigin/lib/CalendarService.ts [hunk 2]
```diff
@@ -172,8 +215,9 @@ export default class BiginCalendarService implements Calendar {
       Event_Title: event.title,
       Start_DateTime: toISO8601String(new Date(event.startTime)),
       End_DateTime: toISO8601String(new Date(event.endTime)),
-      Description: event.additionalNotes,
-      Location: getLocation(event),
+      Description: this.getDescription(event),
+      Venue: getLocation(event),
+      Tag: [{ name: this.eventTag }],
```
[the remark this answers is not in the document]
Comment (sumandas-dev) [reply packages/app-store/zoho-bigin/lib/CalendarService.ts:220]: That's a good idea! but doing so will send 2 emails for the same event to the user. One from Cal.com and another from Bigin. Is that fine?

File: packages/app-store/zoho-bigin/lib/CalendarService.ts (hunk 1 above)
Comment (sumandas-dev) [reply packages/app-store/zoho-bigin/lib/CalendarService.ts:169]: Can you share the steps to reproduce the same? I am getting the description as expected from my side.

**Comment by Udit-takkar** [comment]
@sumandas-dev any update on this PR? is there anything blocking you?

**Comment by keithwillcode** [comment]
Closing this PR due to staleness. Please reopen if work is to continue.

**Pull request closed without merging**